pkg/driver/driver.go

```diff
@@ -8,13 +8,29 @@ import (
 	"github.com/lima-vm/lima/pkg/store"
 )
 
+// Driver interface is used by hostagent for managing vm.
+//
+// This interface is extended by BaseDriver which provides default implementation.
+// All other driver definition must extend BaseDriver
 type Driver interface {
+	// Validate returns error if the current driver isn't support for given config
 	Validate() error
 
+	// CreateDisk returns error if the current driver fails in creating disk
 	CreateDisk() error
 
-	Start(_ context.Context) (chan error, error)
+	// Start is used for booting the vm using driver instance
+	// It returns a bool as first return value, indicating whether StartGUI need to be called
+	// The second argument will be a chan error returned on successful boot
+	// The third argument may contain error occurred while starting driver
+	Start(_ context.Context) (bool, chan error, error)
 
+	// StartGUI is used for starting GUI synchronously by hostagent. This method should be returned only after vm terminates
```

Comment:
```suggestion
	// RunGUI is used for starting GUI synchronously by hostagent. This method should be returned only after vm terminates
```
(Similar to os/exec.Command.Run)

Reply:
Done